thrifty-schema/src/main/java/com/microsoft/thrifty/schema/AbstractUserElementBuilder.java

```diff
@@ -64,5 +65,16 @@ public TBuilder annotations(Map<String, String> annotations) {
         return (TBuilder) this;
     }
 
+    public TBuilder type(TElement type) {
```

Comment:
Is this intended to be a "set all builder params to this `TElement`" kind of function, or do you imagine it to be just constrained to `UserElement` values?  As is, I like it but can imagine surprises down the road:

```
StructType s1 = ...
StructType s2 = ... //something totally different

s1.toBuilder().
    type(s2)  // why?  who knows!  but now the builder is basically s2 with the fields of s1
```